src/cli.ts

```diff
@@ -76,12 +85,38 @@ async function ask(): Promise<Answers> {
 		answers.replay = replayFile;
 	}
 
+	if (!!argv.import) {
+		try {
+			const importDirectory = path.resolve(argv.import);
+			importContext = new ImportContext(importDirectory);
+			await importContext.load();
+		} catch (error) {
+			console.error(error);
+			throw new Error(
+				"Please ensure that --import points to a valid adapter directory",
+			);
+		}
+		if (await importContext.fileExists(".create-adapter.json")) {
+			// it's just not worth trying to figure out things if the adapter was already created with create-adapter
+			throw new Error(
+				"Use --replay instead of --import for an adapter created with a recent version of create-adapter.",
+			);
+		}
+	}
+
 	async function askQuestion(q: Question): Promise<void> {
 		if (testCondition(q.condition, answers)) {
 			// Make properties dependent on previous answers
 			if (typeof q.initial === "function") {
 				q.initial = q.initial(answers);
 			}
+			if (importContext && q.import) {
+				let imported = q.import(importContext, answers, q);
+				if (imported && typeof (imported as any).then === "function") {
```

Comment:
```suggestion
				if (imported instanceof Promise) {
```
maybe?

Reply:
Thanks, was looking for that but apparently used the wrong Google search terms ;-)